internal/scorecard/alpha/tests/test_bundle.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// TestBundle holds the bundle contents to be tested
+type TestBundle struct {
+	PackageManifest registry.PackageManifest
+	BundleErrors    []errors.ManifestResult
+	Bundles         []*registry.Bundle
+	CRs             []unstructured.Unstructured
+}
+
+// GetBundle parses a Bundle from a given on-disk path returning a TestBundle
+func GetBundle(bundlePath string) (cfg TestBundle, err error) {
+
+	validationLogOutput := new(bytes.Buffer)
+	origOutput := logrus.StandardLogger().Out
+	logrus.SetOutput(validationLogOutput)
+	defer logrus.SetOutput(origOutput)
+
+	cfg.PackageManifest, cfg.Bundles, cfg.BundleErrors = manifests.GetManifestsDir(bundlePath)
```

Comment:
I could be wrong, but I think the `bundlePath` will look like this (from the [OLM operator bundle enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md)):
```sh
$ tree
/
├── manifests
│   ├── testbackup.crd.yaml
│   ├── testcluster.crd.yaml
│   ├── testoperator.v0.1.0.clusterserviceversion.yaml
│   └── testrestore.crd.yaml
└── metadata
    └── annotations.yaml
```

I think we need to find a different function that loads a single bundle from disk.

@estroz @kevinrizza @ecordell does something exist that turns a bundle directory like this into a `registry.Bundle`? I don't see anything [here](https://github.com/operator-framework/operator-registry/blob/master/pkg/registry/bundle.go) that seems to help, but I'm probably overlooking something.

If we have that, I think most of this file may go away. We could basically use `registry.Bundle` directly, possibly wrapping it to expose the CRs from the CSVs alm-examples.

Reply:
This function, which is called by an [`SQLPopulator`](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/directory.go#L71) from operator-registry, will return a single `registry.Bundle` if `bundlePath` is a bundle directory, i.e. `len(cfg.Bundles) == 1`.

Reply:
@estroz Does that help us? It seems like we're actually after is something like [`loadBundle`](https://github.com/operator-framework/operator-registry/blob/a6b3b407897530e27294d2f47cefc065c57a2716/pkg/sqlite/directory.go#L180), but for that, it seems like we would pass `/bundle/manifests` and then `/bundle/metadata` would be ignored.

We probably also don't want all the logrus logging.

Reply:
The logging isn't great, and I have a PR upstream to configure it. For now turning it off when running `GetManifestsDir` should be enough. Do we care about metadata in this PR?

Reply:
Maybe not. Just trying to capture the _potential_ issues I see and want to make sure we don't let tech debt creep in from the beginning. Totally fine by me to add a TODO comment and a follow-on task to address this later.

Reply:
I added a TODO around changing the get bundle API call, there is a story to handle the bundle image case so thats probably a good time to address this API.